File: WoodPeckerDemo/WoodPeckerDemo/Modules/Sandbox/SandBoxViewController.m
//
//  SandBoxViewController.m
//  WoodPeckerDemo
//
//  Created by xiaogang zhang on 2018/1/17.
//  Copyright © 2018年 lifebetter. All rights reserved.
//

#import "SandBoxViewController.h"

@interface SandBoxViewController ()

@end

@implementation SandBoxViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)addFileButtonPressed:(id)sender {
    NSDateFormatter * formatter = [[NSDateFormatter alloc] init];
    [formatter setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
    NSDate * date = [NSDate date];
    NSString * text = [formatter stringFromDate:date];
    //file name
    [formatter setDateFormat:@"HH-mm-ss"];
    NSString * fileName = [NSString stringWithFormat:@"%@.txt",[formatter stringFromDate:date]];
    NSString * documentPath = [NSHomeDirectory() stringByAppendingPathComponent:@"Documents"];
    NSString * filePath = [documentPath stringByAppendingPathComponent:fileName];
    NSURL * fileURL = [NSURL fileURLWithPath:filePath];
    NSError * error = nil;
    [text writeToURL:fileURL atomically:YES encoding:NSUTF8StringEncoding error:&error];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];

}


@end

File: WoodpeckerMacDemo/WoodpeckerMacDemo/Pages/Sandbox/SandboxViewController.swift
//
//  SandboxViewController.swift
//  WoodpeckerMacDemo
//
//  Created by xiaogang zhang on 2019/6/1.
//  Copyright © 2019 LIFE BETTER. All rights reserved.
//

import Cocoa

class SandboxViewController: NSViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
   
    }
    
    @IBAction func addButtonClicked(_ sender: Any) {
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
        let date = Date()
        let text = formatter.string(from: date)
        //file name
        formatter.dateFormat = "HH-mm-ss"
        let fileName = "\(formatter.string(from: date)).txt"
        let documentPath = (NSHomeDirectory() as NSString).appendingPathComponent("Documents")
        let filePath = (documentPath as NSString).appendingPathComponent(fileName)
        let fileURL = NSURL.fileURL(withPath: filePath)
        
        do {
            try text.write(to: fileURL, atomically: true, encoding: String.Encoding.utf8)
        } catch  {
            
        }
    }
    
}
